lib/msf/core/auxiliary/ubiquiti.rb

```diff
@@ -0,0 +1,336 @@
+    def process_radiusprofile(lines, credential_data)
+      lines.each do |line|
+        line['auth_servers'].each do |server|
+          report_service(
+            host: server['ip'],
+            port: server['port'],
+            name: 'radius',
+            proto: 'udp'
+          )
+          if server['x_secret'] # no need to output if the secret is blank, therefore its not configured
+            print_good("Radius server: #{server['ip']}:#{server['port']} with secret '#{server['x_secret']}'")
+            cred = credential_data.dup
+            cred[:username] = ''
+            cred[:private_data] = server['x_secret']
+            cred[:private_type] = :password
+            cred[:address] = server['ip']
+            cred[:port] = server['port']
+            create_credential_and_login(cred)
+          end
+        end
+      end
+    end
+
+    # settings has multiple items we care about:
+    #   x_mesh_essid/x_mesh_psk -> should contain the mesh network wifi name and password
+    #   ntp -> ntp servers
+    #   x_ssh_username/x_ssh_password/x_ssh_keys/x_ssh_sha512passwd
+
+    # Example lines
+    # {"__cmd"=>"select", "collection"=>"setting"}
+    # {"_id"=>BSON::ObjectId('3c3e21ac3715ce20a721d9ba'), "site_id"=>"3c2f215b3825ca2087c1dfb6", "key"=>"ntp", "ntp_server_1"=>"0.ubnt.pool.ntp.org", "ntp_server_2"=>"1.ubnt.pool.ntp.org", "ntp_server_3"=>"2.ubnt.pool.ntp.org", "ntp_server_4"=>"3.ubnt.pool.ntp.org"}
+    # {"_id"=>BSON::ObjectId('3c3e21ac3715ce20a721d9bb'), "site_id"=>"3c2f215b3825ca2087c1dfb6", "key"=>"mgmt", "advanced_feature_enabled"=>false, "x_ssh_enabled"=>true, "x_ssh_bind_wildcard"=>false, "x_ssh_auth_password_enabled"=>true, "unifi_idp_enabled"=>true, "x_mgmt_key"=>"ba6cbe170f8276cd86b24ac79ab29afc", "x_ssh_username"=>"admin", "x_ssh_password"=>"16xoB6F2UyAcU6fP", "x_ssh_keys"=>[], "x_ssh_sha512passwd"=>"$6$R4qnAaaF$AAAlL2t.fXu0aaa9z3uvcIm3ujbtJLhIO.lN1xZqHZPQoUAXs2BUTmI5UbuBo2/8t3epzbVLib17Ls7GCVx7V."}
+    # {"_id"=>BSON::ObjectId('3c3e21ac3715ce20a721d9bc'), "site_id"=>"3c2f215b3825ca2087c1dfb6", "key"=>"connectivity", "enabled"=>true, "uplink_type"=>"gateway", "x_mesh_essid"=>"vwire-851237d214c8c6ba", "x_mesh_psk"=>"523a9b872b4624c7894f96c3ae22cdfa"}
+    # {"_id"=>BSON::ObjectId('3c3e21ac3715ce20a721d9bd'), "site_id"=>"3c2f215b3825ca2087c1dfb6", "key"=>"snmp", "community": "public", "enabled": true, "enabledV3": true, "username": "usernamesnmpv3", "x_password": "passwordsnmpv3"}
+    def process_settings(lines, credential_data)
+      lines.each do |line|
+        case line['key']
+        when 'snmp'
+          unless line['community'].blank?
+            cred = credential_data.dup
+            cred[:protocol] = 'udp'
+            cred[:port] = 161
+            cred[:service_name] = 'snmp'
```

Comment:
You could bump these 4 lines out of this block and up a level so they don't need to be repeated when the line is determined to be an SNMPv3 entry.